### ISSUE TITLE
How to catch the exception "League\Flysystem\FileNotFoundException"

### ISSUE THREAD

**Issue by slince** [ADmad/cakephp-glide#18]
I want send a default image if the client requests an image  that does not exist.

**Comment by slince** [comment]
It should not raise a 500 error.

**Comment by ADmad** [comment]
There are a few options:

1. You can set `ignoreException` to `false` in which case the middleware won't throw any exception if it's unable to generate a response. Then use your own middleware to check for required URL pattern and return your default image. 

2. Extend the middlware and override `_getResponse()`. Wrap `parent::_getResponse()` in try catch block and handle exception as required.

3. Submit a PR to add `exceptionCallback` option which will be called in case of exception with the exception object as argument. The callback can then return required response object.

**Comment by slince** [comment]
It is the ability to solve the problem, but not flexible enough; How about using the event mechanism ?

```php

$middleware->on(Events::EXCEPTION_RAISED, function($event){
    $event->setResponse(new Response('No images found‘));
});




// in GlideMiddlewar
protected function _getResponse(ServerRequestInterface $request, ResponseInterface $response, Server $server)
{
    try {
        $response = $server->getImageResponse($this->_path, $this->_params);
    } catch (\Exception $e) {
         $event = new Event(Events::EXCEPTION_RAISED, $this, [
                'exception' => $e
         ]);
         $this->eventManager()->dispatch($event);
         if  ($response = $event->getResponse()) {
              return $response;
         }
         if  ($event->isIgnoreException())  {
             return null;
         }
         throw $e;
    }

    return $response;
}

```

**Comment by ADmad** [comment]
A PR with proposed change is welcome.

**Comment by ADmad** [comment]
`$event->setResponse()\getReponse()` isn't necessary. Can just use existing `Event::setResult()/getResult()`.

**Comment by slince** [comment]
@ADmad  Ok, i will try to create a pr for this.

**Comment by slince** [comment]
@ADmad   However maybe `setResponse()\getReponse()` has better readability.

**Comment by slince** [comment]
Hi
 i have created a pr #19;

**Comment by ADmad** [comment]
Fixed in 3.0.0
